File: algorithm/leetcode5420.final-prices-with-a-special-discount-in-a-shop.cpp
class Solution {
public:
    vector<int> finalPrices(vector<int>& prices) {
        int i, j, n = prices.size();
        for(i = 0; i < n-1; i++)
        {
            for(j = i+1; j < n; j++)
            {
                if(prices[j] <= prices[i])
                {
                    prices[i] -= prices[j];
                    break;
                }
            }
        }
        return prices;
    }
};

File: algorithm/Python3_version/leetcode5420.final-prices-with-a-special-discount-in-a-shop.py
class Solution:
    def finalPrices(self, prices: List[int]) -> List[int]:
        n = len(prices)
        for i in range(n-1):
            for j in range(i+1,n):
                if prices[j] <= prices[i]:
                    prices[i] -= prices[j]
                    break;
        return prices